ShinyExpenseTrackerTests/src/ca/ualberta/cs/shinyexpensetracker/test/ViewAllExpenseClaimsActivityTests.java

```diff
@@ -144,59 +144,100 @@ public void testAddedClaimIsVisible() {
 	/* All of the testClaimsSorted check roughly the same thing,
 	 * going through all 6 permutations of {new, mid, old}.
 	 */
-	public void testClaimsSortedA() {
-		ExpenseClaim newClaim = addClaim(new ExpenseClaim("New Claim", new Date(1000000)));
-		ExpenseClaim midClaim = addClaim(new ExpenseClaim("Mid Claim", new Date(300000)));
-		ExpenseClaim oldClaim = addClaim(new ExpenseClaim("Old Claim", new Date(10000)));
+	public void testClaimsSorted() throws Exception {
+		ExpenseClaim[] testingClaims = {
+				new ExpenseClaim("Old Claim", new Date(1000)),
+				new ExpenseClaim("Mid Claim", new Date(2000)),
+				new ExpenseClaim("New Claim", new Date(3000)),
+		};
+		int numTests = 0;
 
-		assertEquals("new claim must come first in the list", newClaim, getClaim(0));
-		assertEquals("mid claim must be in the middle of the list", midClaim, getClaim(1));
-		assertEquals("old claim must come last in the list", oldClaim, getClaim(2));
-	}
-	public void testClaimsSortedB() {
-		ExpenseClaim newClaim = addClaim(new ExpenseClaim("New Claim", new Date(1000000)));
-		ExpenseClaim oldClaim = addClaim(new ExpenseClaim("Old Claim", new Date(10000)));
-		ExpenseClaim midClaim = addClaim(new ExpenseClaim("Mid Claim", new Date(300000)));
+		// Check that our test case items compare correctly
+		// -- Equality Cases
+		assertEquals(0, testingClaims[0].compareTo(testingClaims[0]));
+		assertEquals(0, testingClaims[1].compareTo(testingClaims[1]));
+		assertEquals(0, testingClaims[2].compareTo(testingClaims[2]));
+		// -- Inequality Cases
+		assertEquals(-1, testingClaims[0].compareTo(testingClaims[1]));
+		assertEquals(-1, testingClaims[0].compareTo(testingClaims[2]));
+		assertEquals(-1, testingClaims[1].compareTo(testingClaims[2]));
+		// -- Greater-than tests are done in the loop.
 
-		assertEquals("new claim must come first in the list", newClaim, getClaim(0));
-		assertEquals("mid claim must be in the middle of the list", midClaim, getClaim(1));
-		assertEquals("old claim must come last in the list", oldClaim, getClaim(2));
-	}
-	public void testClaimsSortedC() {
-		ExpenseClaim midClaim = addClaim(new ExpenseClaim("Mid Claim", new Date(300000)));
-		ExpenseClaim newClaim = addClaim(new ExpenseClaim("New Claim", new Date(1000000)));
-		ExpenseClaim oldClaim = addClaim(new ExpenseClaim("Old Claim", new Date(10000)));
+		// i, j, k go to 0.
+		int i = 3;
+		while (i --> 0) {
```

Comment:
But really, as funny as the "goes to" operator is, could you change these for `for` loops? What you're doing would be clearer at a glance.